src/gvmd.c

```diff
@@ -2660,6 +2656,31 @@ gvmd (int argc, char** argv)
   if (gvm_auth_init ())
     exit (EXIT_FAILURE);
 
+  /* Try to get OSP VT update socket from default OpenVAS if it
+   *  was not set with the --osp-vt-update option.
+   */
+  if (get_osp_vt_update_socket () == NULL)
+    {
+      char *default_socket = openvas_default_scanner_host ();
+      if (default_socket)
+        {
+          g_debug ("%s: Using OSP VT update socket from default OpenVAS"
+                   " scanner: %s",
+                   __FUNCTION__,
+                   default_socket);
+          set_osp_vt_update_socket (default_socket);
+        }
+      else
+        {
+          g_critical ("%s: No OSP VT update socket found."
```

Comment:
I'm OK with merging it this way now, but I wonder if it wouldn't be better to check this every time the update is scheduled (and just give a warning if there's no socket path).  Then the user would be able to update the default scanner via GMP, and would be able to start gvmd without needing to have this set up correctly.

Reply:
Would changing the line to a `g_warning` and removing the following `return EXIT_FAILURE;` produce the behavior you'd prefer?

Reply:
It wouldn't be enough, we would also have to try to get the OSP VT update socket in update_nvt_cache_retry if it is not already set.

Reply:
Let's leave it for now.  Lots of other things to do.